### ISSUE TITLE
[TST] Tests should have docstrings that reveal their intent

### ISSUE THREAD

**Issue by ericmjl** [ericmjl/pyjanitor#306]
The test docstring should reveal what they are intending to test, in plain English. Even if the name is already very informative, having the plain English comment right next to it and in context would very much help with maintenance.

**Comment by hectormz** [comment]
I might take this up and workon piecewise

**Comment by hectormz** [comment]
@ericmjl I've been using [`interrogate`](https://interrogate.readthedocs.io/en/latest/) lately to find tests (or any functions) without docstrings. Just putting it on your radar if you're not familiar as an unofficial/official tool to check for docstrings. Especially once we get to 100% docstring coverage.

**Comment by ericmjl** [comment]
That's a wonderful find, @HectorM14! One more thing to automate out :smile:. If I don't get to it before you (I'll probably do it later this afternoon, so I don't get sucked out of work), feel free to PR in, otherwise hang tight and I'll make the PR later in the evening!

**Comment by hectormz** [comment]
I'm glad you like it. I found it via this aptly named blog post: ["Why You Should Document Your Tests"](https://hynek.me/articles/document-your-tests/). The post provides a pattern to run `interrogate` specifically on tests:

```bash
$ interrogate -vv --fail-under 100 --whitelist-regex "test_.*" tests
```

I'm not sure how you'd like to implement this. Perhaps in `pre-commit`/`make`/`CI`?

For now it might be worth separating tests and non-tests. With the flags I'm currently using: 
`$ interrogate -vv --fail-under 100 -s -I janitor\`
non-tests have 85.4% coverage, and tests have 13% coverage.

The missing non-test docstrings seem to largely be module-level docstrings, and nested functions (which perhaps should have docstrings anyway, but there is no flag currently to ignore these). 

We also don't always have docstrings for semi-private functions (leading with underscore), but perhaps those should at least have a simple docstring as well.

**Comment by ericmjl** [comment]
@HectorM14 let's place it as part of the code style checking functions in the Makefile, so that this gets executed in CI. Pre-commit also sounds reasonable as well. I ended up not getting to it haha, so please feel free to start the PR. I should really dedicate my Fridays to doing OSS work.

**Comment by hectormz** [comment]
@ericmjl Sure, I can do that! I did notice that the output is not formatted correctly for Windows. I'm working on a pull request for that. And once that's settled, I'll introduce a PR here!
